libz-rs-sys-cdylib/README.md

```diff
@@ -74,6 +76,39 @@ the standard library has the following limitations:
 
 - CPU feature detection is currently disabled. This is true for both compile-time and run-time feature detection.
     This means `zlib-rs` will not make use of SIMD or other custom instructions.
-- The `rust-allocator` should not be used. It internally enables the standard library, causing issues. Using `c-allocator` 
-    or not providing an allocator at build time is still supported.On embedded it is most common to provide a custom allocator 
+- The `rust-allocator` should not be used. It internally enables the standard library, causing issues. Using `c-allocator`
+    or not providing an allocator at build time is still supported.On embedded it is most common to provide a custom allocator
     that "allocates" into a custom array.
+
+## Build for Distribution
+
+A `cargo build` currently does not set some fields that are required or useful when using a dynamic library from C.
+For instance, the soname and version are not set by a standard `cargo build`.
+
+To build a proper, installable dynamic library, we recommend [`cargo-c`](https://github.com/lu-zero/cargo-c):
+
+```
+cargo install cargo-c
+```
+
+This tool deals with setting fields (soname, version) that a normal `cargo build` does not set (today).
+It's configuration is in the `Cargo.toml`, where e.g. the library name or version can be changed.
+
+```
+> cargo cbuild --release
+   Compiling zlib-rs v0.2.1
+   Compiling libz-rs-sys v0.2.1
+   Compiling libz-rs-sys-cdylib v0.2.1
+    Finished `release` profile [optimized] target(s) in 1.86s
+    Building pkg-config files
+> tree target
+target
+├── CACHEDIR.TAG
+└── x86_64-unknown-linux-gnu
+    └── release
+        ├── libz_rs.a
+        ├── libz_rs.d
+        ├── libz_rs.pc
+        ├── libz_rs.so
+        └── libz_rs-uninstalled.pc
```

Comment:
@lu-zero is there a reason `cargo cbuild` puts its things in `target/<target tripple>/release` rather than just `target/release`? it's annoyingly inconsistent with a standard `cargo build`.

Reply:
It is required until cargo learns to differentiate the RUSTFLAGS for the build.rs and the RUSTFLAGS for the host.

using `cargo cbuild -v` it emits a note with the uninstalled pkg-config path.

Reply:
https://github.com/rust-lang/cargo/issues/4423 is another way to see the problem and https://github.com/rust-lang/cargo/issues/9452 is how to solve it.

But normally `cinstall` is used to produce the `.a` that is then linked in the bigger C project. Even C projects support cross compilation ^^.

I guess I should really prepare a blogpost showing how to use cargo-c in many deployment scenario :)